uiowa/uiowa
# Require Authentication - Restrict IP method

Resolves https://github.com/uiowa/uiowa/issues/4241
Resolves #4242

Currently relying on a patch from https://www.drupal.org/project/restrict_ip/issues/2957482, Cory has been working on test coverage to get that fix in sooner.

## To Do

- [x] Freshen up access denied page like https://github.com/uiowa/uiowa/issues/4241#issuecomment-1006894018
- [x] Cache-Control headers set to no-cache? Test in DEV?
- [x] Switch to stable patch file or update module (pending merge)
- [x] Disable search from header
- [x] Private files
- [x] Write test for private files

## Later on

- See https://github.com/uiowa/uiowa/issues/4640

## To Test
- `blt drupal:install --site default`
- `drush en sitenow_intranet`
- `drush cim`
- Non-admins should get an access denied on all drupal paths except for the default content public files. These will be removed manually before delivering to initial customers.
- Run the private files scheme test: `ddev phpunit tests/phpunit/`
  - Try manually deleting one of the file related patch files and run again.
  - Try changing one of the patch files' values from private to something else and run again.
